pilot/pkg/model/jwks_resolver.go

```diff
@@ -414,9 +440,31 @@ func (r *JwksResolver) refresh() {
 			r.PushFunc()
 		}
 	}
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	var useDefault bool
+	if hasErrors {
```

Comment:
are we setting more aggressive retry interval for ALL endpoints if a single one fails?

Reply:
Yes you are right... This isn't that great. I can alter the code to only refresh more aggressively the failing endpoints. I was also thinking about if each endpoint should be refreshed with its own interval (spread them with some random time at init) so that they do not synchronously update all at once e.g. if there was some network issue which affected all endpoints. How does that sound? Definitely would need some more work for this but might be better ;)

Reply:
@howardjohn The `PushFunc` is only invoked if there is a change of the key. Even in the error case it will not call the `PushFunc` unless the already cached key is getting evicted (currently 24h) or there is a change of the key. I'm not so familiar with the rotation of jwks_uri but to me it looks like it is not an event which happens often e.g. okta rotates 4 times a year. Therefore I'm curios in what scenario this can be an issue maybe I'm missing something.

Reply:
That said shall I add a unittest to show how often the `PushFunc` is getting called so that I can show that it doesn't get called if not needed? @yangminzhu What do you think about this?